crates/scheduler/src/call/platform/update.rs

```diff
@@ -86,14 +85,13 @@ fn process_<B: Board>(mut call: SchedulerCall<B, api::process::Sig>) {
     let memory = scheduler.applet.memory();
     let result = try {
         let chunk = memory.get(*ptr, *len)?;
-        board::platform::Update::<B>::process(chunk)
+        board::platform::Update::<B>::process(chunk)?
     };
     call.reply(result);
 }
 
 #[cfg(feature = "board-api-platform-update")]
 fn finalize<B: Board>(call: SchedulerCall<B, api::finalize::Sig>) {
     let api::finalize::Params {} = call.read();
-    let result = try { board::platform::Update::<B>::finalize() };
-    call.reply(result);
+    call.reply(board::platform::Update::<B>::finalize().map_err(|x| x.into()));
```

Comment:
nit: this could be simplified to
```suggestion
    call.reply(try { board::platform::Update::<B>::finalize()? });
```

This should work everywhere this pattern happens except when the never type `!` is used. I've created a compiler bug here: https://github.com/rust-lang/rust/issues/125364